sematic/scheduling/tests/test_job_scheduler.py

```diff
@@ -77,10 +73,10 @@ def resolution(run):
 def test_schedule_run(mock_k8s, run: Run, resolution: Resolution):
     resolution.status = ResolutionStatus.RUNNING
     run.container_image_uri = resolution.container_image_uri
-    scheduled = schedule_run(run, resolution)
-    assert len(scheduled.external_jobs) == 1
-    external_job = scheduled.external_jobs[0]
-    assert isinstance(external_job, KubernetesExternalJob)
+    _, jobs = schedule_run(run, resolution, [])
+    assert len(jobs) == 1
+    job = jobs[0]
+    assert isinstance(job, Job)
```

Comment:
Are there any other asserts that can be made on the `job` contents?
Maybe add test cases with mocked failure modes and assert on those, without duplicating too much logic or coverage against the `test_kubernetes.py` `PodSummary` unit tests?

Reply:
added an assert that the job was in the requested state, which is the only thing we really care about from the perspective of testing the scheduler. The other details would be testing k8s logic rather than the job scheduler.